content/guest-post/session-based-authentication-with-nestjs-and-mongodb/index.md

```diff
@@ -0,0 +1,977 @@
+---
+title: "Session Based Authentication with NestJS and MongoDB"
+date: "2022-03-15"
+coverImage: ""
+author: "Ekekenta Odionyenfe Clinton"
+tags: ["Authentication", "NestJS", "MongoDB", "Node.js"]
+description: ""
+---
+
+## Introduction
+
+It is an indisputable reality that authentication is critical in any application or system if you want to secure user data and enable protected access to information. Authentication is the procedure of establishing or demonstrating that something is true, legitimate, or valid.
+
+## Prerequisites
+
+This tutorial is a hands-on demonstration. To follow along, ensure you have the following in place:
+
+- [Node.js](https://nodejs.org/en/) running in your system because NestJS is a Node.js framework
+- [MongoDB](https://www.mongodb.com/) installed
+
+## What is NestJS?
+
+Nest (NestJS) is a Node.js server-side application framework for building scalable, efficient applications.
+
+It is written in TypeScript and built on Express, a very minimalistic framework that is great on its own but lacks structure. It combines the programming paradigms such as object-oriented programming, functional programming, and functional reactive programming.
+
+It is a framework to use if you want a lot of structure on your backend. Its syntax and structure are very similar to AngularJS, a front-end framework. And it uses TypeScript, services, and dependency injection in the same way that AngularJS does.
+
+It employs modules and controllers, and you can build controllers for a file using the command-line interface.
+
+NestJS modules allow you to group related controllers and service providers into a single code file. Simply put, a NestJS module is a TypeScript file with the **@Module** annotation (). This decorator informs the NestJS framework about which controllers, service providers, and other associated resources will be instantiated and used by the app code later.
```

Comment:
@icode247 It is not clear what the parentheses `()` mean here: ```a TypeScript file with the **@Module** annotation ().```

Can you pls clarify a bit?